gcp/table_gcp_logging_bucket.go

```diff
@@ -0,0 +1,180 @@
+			},
+			{
+				Name:        "locked",
+				Description: "Locked: Whether the bucket has been locked. The retention period on a locked bucket may not be changed. Locked buckets may only be deleted if they are empty.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "retention_days",
+				Description: "Logs will be retained by default for this amount of time, after which they will automatically be deleted.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "update_time",
+				Description: "he last update timestamp of the bucket.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromGo().NullIfZero(),
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(bucketAka),
+			},
+
+			// standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(bucketLocation),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getProject,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listLoggingBuckets(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listLoggingBuckets")
+
+	// Create service connection
+	service, err := LoggingService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	project := projectData.Project
+
+	resp := service.Projects.Locations.Buckets.List("projects/" + project + "/locations/-")
+	if err := resp.Pages(
+		ctx,
+		func(page *logging.ListBucketsResponse) error {
+			for _, bucket := range page.Buckets {
+				d.StreamListItem(ctx, bucket)
+			}
+			return nil
+		},
+	); err != nil {
+		return nil, err
+	}
+
+	return nil, err
+}
+
+//// HYDRATED FUNCTIONS
+
+func getLoggingBucket(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getLoggingBucket")
+
+	// Create Service Connection
+	service, err := LoggingService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	bucketName := d.KeyColumnQuals["name"].GetStringValue()
+
+	// Match the bucket name pattern if we are doing query with where clause by passing name value
+	matched, err := regexp.MatchString(`projects/.+/locations/.+/buckets/.+`, bucketName)
+
+	if !matched {
+		return nil, errors.New("Bucket name should match 'projects/[PROJECT_ID]/locations/[LOCATION_ID]/buckets/[BUCKET_ID]'")
+	}
```

Comment:
@ParthaI Can we just provide the shorter name as input when doing GET (i.e. `test123`, not `projects/abc/locations/us-east1/buckets/test123`)?
Since we can form the full name explicitly, before calling the API.